File: questao01_escada/java/src/escada.java
import javax.swing.JOptionPane;

public class escada
{
	public static void main(String[] args)
	{
		int linha, altura, numero;
		int sair = 1;

		System.out.println("\nSUA ESCADA SERÁ IMPRESSA AQUI!");

		while (true)
		{
			try
			{
				numero = Integer.parseInt(JOptionPane.showInputDialog("Por favor, leia a instruções abaixo:"
						+ "\n*Digite na caixa de mensagens somente números inteiros*"
						+ "\n*Sua escada será impressa no terminal*\n\n"
						+ "Informe o tamanho da escada ou digite '0' para sair"));// Box que verifica se o número é
																					// interiro.
				// Não possibilita a entrada de número negativos
				if (numero > -1)
				{
					sair = numero;
					// Condição para não sair do programa.
					if (sair != 0)
					{
						// Serve para acrescentrar altura.
						for (altura = 1; altura <= numero; altura++)
						{
							// Faz a linha.
							for (linha = 1; linha <= numero; linha++)
							{
								// Verifica o que vai ser acrescentado. Espaço ou asterisco.
								if (linha > (numero - altura))
								{
									System.out.print("*");
								} else
								{
									System.out.print(" ");
								}
							}
							System.out.println();
						}
					} else
					{// Condição para sair do programa.
						System.out.println("\nSaindo...");
						break;
					}
				} else
				{// Caso seja inserido algum número negativo.
					System.out.println("\nPor favor, digite somente números positivos!");
				}
			} catch (NumberFormatException exception)
			{// Caso o digito inserido não seja um número inteiro é
				// retornado esse erro.
				// caso tenha sido digitado uma String a exceção é capturada
				JOptionPane.showMessageDialog(null, "Error: Por favor, informe somente números inteiros!");
			}
		}
	}
}

File: questao01_escada/python/escada.py
import sys

print("{}\nPor favor, leia a instruções abaixo:\n*Digite somente números inteiros.\n"
"*Para fechar o programa digite 'sair' ou 's'.\n{}\n".format(("#" * 50), ("#" * 50)))

while True:
    lista_space = []
    lista_asterisco = []
    linha = 0
    try:
        degraus = input("Digite a quantidade de degraus que sua escada vai ter: ")
        sair = str(degraus).upper()
        if sair == "SAIR" or sair == "S":
            print("\nSaindo...\n")
            break
        else:
            degraus = int(degraus)
            while linha <= degraus:
                lista_space.append(" ")
                linha = linha + 1
            while degraus > 0:
                lista_asterisco.append("*")
                lista_space.pop(0)
                degraus = degraus - 1
                lista =  lista_space + lista_asterisco
                #lista2.reverse()
                for j in lista:   
                    sys.stdout.write(j)
                print()
    except:
        print("Digite somente números inteiros!")
